e2e-tests/development-runtime/package.json

```diff
@@ -0,0 +1,51 @@
+{
+  "name": "e2e-development-runtime",
+  "description": "An e2e-test validating development runtime functionality",
+  "version": "1.0.0",
+  "author": "Dustin Schau <dustin@gatsbyjs.com>",
+  "dependencies": {
+    "gatsby": "^2.0.71",
+    "gatsby-image": "^2.0.20",
+    "gatsby-plugin-manifest": "^2.0.9",
+    "gatsby-plugin-offline": "^2.0.20",
+    "gatsby-plugin-react-helmet": "^3.0.2",
+    "gatsby-plugin-sharp": "^2.0.14",
+    "gatsby-source-filesystem": "^2.0.12",
+    "gatsby-transformer-remark": "^2.1.17",
+    "gatsby-transformer-sharp": "^2.1.8",
+    "prop-types": "^15.6.2",
+    "react": "16.7.0-alpha.2",
+    "react-dom": "16.7.0-alpha.2",
+    "react-helmet": "^5.2.0"
+  },
+  "keywords": [
+    "gatsby",
+    "gatsby-e2e"
+  ],
+  "license": "MIT",
+  "scripts": {
+    "build": "gatsby build",
+    "develop": "gatsby develop",
+    "serve": "gatsby serve",
+    "start": "npm run develop",
+    "format": "prettier --write \"src/**/*.js\"",
+    "test": "npm run start-server-and-test || npm run reset",
```

Comment:
Does this only reset if the tests fail? Will `posttest` always be run anyway?

Would removing `posttest` and changing this to `"npm run reset && npm run start-server-and-test"` do the same thing? 

(asking because I had to look up what `||` does in bash)

Reply:
Both! `||` should handle the failure case.

Unfortunately, `posttest` (or any `post-` script) doesn't get invoked in a failure case, e.g. the following will not invoke `posttest`

```json
{
  "name": "whatever",
  "version": "1.0.0",
  "main": "index.js",
  "license": "MIT",
  "scripts": {
    "test": "exit 1",
    "posttest": "echo \"I will not be triggered\""
  }
}
```